### PR TITLE
Ignore null-values when encoding

### PR DESCRIPTION
**Pull request by jhermsmeier**
**Description**

Currently, when encoding objects of any kind, values that are either `null` or `undefined` will throw `TypeError: Cannot read property 'constructor' of null` because we don't even check for those.

As it can become very cumbersome to remove any keys (or items) with null-values when working with more complex data structures, I propose ignoring null-values during encoding, so that they'll be left out in the encoded data.

This change should theoretically be backwards compatible, if I haven't missed anything.

**Changes**

- `bencode.decode()` now returns `null` if fed with empty values (`''` or `new Buffer(0)`)
- `bencode.encode()` and parts now skip / ignore `null` and `undefined` in Arrays and Objects
- `bencode.encode()` now returns an empty Buffer / String when invoked on `null` or `undefined`

**Example**

```js
bencode.encode([
  { empty: null },
  { notset: undefined },
  null,
  undefined,
  0
])
```
Will now encode to the following (instead of throwing):
```js
[ {}, {}, 0 ]
```

### CODE REVIEW

**Comment by themasch** [comment]
I was thinking about allowing `null` and just ignoring `undefined` but since theres no encoding for `null`-values in bencode I think this is fine. :+1:

File: package.json [hunk 1]
```diff
@@ -42,7 +42,8 @@
   },
   "scripts": {
     "bench": "matcha",
-    "test": "standard && tape test/*.test.js | tap-spec"
+    "test": "standard && tape test/*.test.js | tap-spec",
+    "test:new": "tape null-values.test.js | tap-spec"
```
Comment (themasch) [inline package.json]: I had to add a "test/" in front of "null-values.test.js" here because tape wasn't able to find the test case. Can you reproduce that?

Comment (jhermsmeier) [inline package.json]: Oh boy, I forgot to remove that again – just wanted to use it to run that particular one in quick succession without the noisy output from the other tests, but then I never did
